### ISSUE TITLE
Uncaught TypeError: Cannot read property 'toLowerCase' of undefined

### ISSUE THREAD

**Issue by xiongxiong** [atom/atom#13154]
[Enter steps to reproduce below:]

1. ...
2. ...

**Atom Version**: 1.12.0-beta7
**Electron Version**: 1.3.6
**System**: Mac OS X 10.12.1
**Thrown From**: Atom Core

### Stack Trace

Uncaught TypeError: Cannot read property 'toLowerCase' of undefined

```
At /Applications/Atom Beta.app/Contents/Resources/app.asar/node_modules/atom-keymap/lib/helpers.js:223

TypeError: Cannot read property 'toLowerCase' of undefined
    at exports.keystrokeForKeyboardEvent (/Applications/Atom Beta.app/Contents/Resources/app.asar/node_modules/atom-keymap/lib/helpers.js:223:18)
    at KeymapManager.module.exports.KeymapManager.keystrokeForKeyboardEvent (/Applications/Atom Beta.app/Contents/Resources/app.asar/node_modules/atom-keymap/lib/keymap-manager.js:430:14)
    at KeymapManager.module.exports.KeymapManager.handleKeyboardEvent (/Applications/Atom Beta.app/Contents/Resources/app.asar/node_modules/atom-keymap/lib/keymap-manager.js:318:24)
    at WindowEventHandler.module.exports.WindowEventHandler.handleDocumentKeyEvent (/Applications/Atom Beta.app/Contents/Resources/app.asar/src/window-event-handler.js:106:36)
    at HTMLDocument.<anonymous> (/Applications/Atom Beta.app/Contents/Resources/app.asar/src/window-event-handler.js:3:59)
```

### Commands

```
     -0:48 vim-mode:insert-below-with-newline (atom-text-editor.editor.vim-mode.normal-mode.is-focused)
 16x -0:40.4.0 core:backspace (atom-text-editor.editor.vim-mode.is-focused.insert-mode)
     -0:00.1.0 find-and-replace:show-replace (atom-text-editor.editor.vim-mode.is-focused.insert-mode)
```

### Config

```json
{
  "core": {
    "customFileTypes": {
      "source.ini": [
        ".buckconfig",
        ".flowconfig"
      ],
      "source.json": [
        ".arcconfig",
        "BUCK.autodeps"
      ],
      "source.python": [
        "BUCK"
      ]
    },
    "disabledPackages": [
      "spell-check",
      "language-csharp",
      "language-gfm"
    ],
    "telemetryConsent": "limited",
    "themes": [
      "atom-light-ui",
      "atom-light-syntax"
    ]
  },
  "editor": {
    "fontFamily": "Mononoki",
    "fontSize": 16,
    "invisibles": {},
    "tabType": "soft"
  }
}
```

### Installed Packages

```coffee
# User
atom-beautify, v0.29.13 (inactive)
atom-terminal, v0.8.0 (inactive)
autocomplete-go, v1.3.0 (active)
builder-go, v1.0.1 (active)
dash, v1.7.0 (inactive)
docblockr, v0.8.7 (active)
go-config, v1.2.4 (active)
go-debug, v0.1.22 (active)
go-get, v2.1.1 (active)
go-imports, v0.1.1 (inactive)
godoc, v1.1.1 (active)
gofmt, v1.2.0 (active)
gometalinter-linter, v1.2.2 (active)
gorename, v1.1.0 (active)
language-asciidoc, v1.5.3 (active)
language-cpp14, v0.6.2 (active)
language-dot, v1.6.0 (active)
language-gradle, v0.0.3 (active)
language-groovy, v0.6.0 (active)
language-haskell, v1.9.12 (active)
language-markdown, v0.18.1 (active)
language-plantuml, v0.2.0 (active)
language-swift, v0.4.0 (active)
linter, v1.11.18 (active)
navigator-go, v1.1.3 (active)
project-manager, v3.3.3 (active)
script, v3.11.1 (inactive)
tester-go, v2.0.2 (active)
vim-mode, v0.65.1 (active)
atom-dark-syntax, v0.27.0 (inactive)
atom-dark-ui, v0.52.0 (inactive)
atom-light-syntax, v0.28.0 (active)
atom-light-ui, v0.45.0 (active)
base16-tomorrow-dark-theme, v1.3.0 (inactive)
base16-tomorrow-light-theme, v1.3.0 (inactive)
one-dark-ui, v1.6.2 (inactive)
one-light-ui, v1.6.2 (inactive)
one-dark-syntax, v1.5.0 (inactive)
one-light-syntax, v1.5.0 (inactive)
solarized-dark-syntax, v1.0.5 (inactive)
solarized-light-syntax, v1.0.5 (inactive)
about, v1.7.0 (active)
archive-view, v0.62.0 (active)
autocomplete-atom-api, v0.10.0 (active)
autocomplete-css, v0.13.1 (active)
autocomplete-html, v0.7.2 (active)
autocomplete-plus, v2.31.4 (active)
autocomplete-snippets, v1.11.0 (active)
autoflow, v0.27.0 (inactive)
autosave, v0.23.1 (active)
background-tips, v0.26.1 (active)
bookmarks, v0.42.0 (active)
bracket-matcher, v0.82.2 (active)
command-palette, v0.39.0 (inactive)
deprecation-cop, v0.54.1 (active)
dev-live-reload, v0.47.0 (active)
encoding-selector, v0.22.0 (active)
exception-reporting, v0.40.0 (active)
find-and-replace, v0.202.0 (inactive)
fuzzy-finder, v1.4.0 (active)
git-diff, v1.1.0 (active)
go-to-line, v0.31.0 (inactive)
grammar-selector, v0.48.2 (active)
image-view, v0.60.0 (active)
incompatible-packages, v0.26.1 (active)
keybinding-resolver, v0.35.0 (active)
line-ending-selector, v0.5.0 (active)
link, v0.31.2 (inactive)
markdown-preview, v0.158.8 (active)
metrics, v1.1.1 (active)
notifications, v0.65.1 (active)
open-on-github, v1.2.1 (inactive)
package-generator, v1.0.1 (inactive)
settings-view, v0.243.1 (active)
snippets, v1.0.3 (active)
spell-check, v0.68.4 (inactive)
status-bar, v1.4.1 (active)
styleguide, v0.47.2 (active)
symbols-view, v0.113.1 (inactive)
tabs, v0.103.0 (active)
timecop, v0.33.2 (active)
tree-view, v0.210.0 (active)
update-package-dependencies, v0.10.0 (active)
welcome, v0.35.1 (active)
whitespace, v0.35.0 (active)
wrap-guide, v0.38.2 (active)
language-c, v0.54.0 (active)
language-clojure, v0.22.1 (active)
language-coffee-script, v0.48.0 (active)
language-csharp, v0.12.1 (inactive)
language-css, v0.40.1 (active)
language-gfm, v0.88.0 (inactive)
language-git, v0.15.0 (active)
language-go, v0.43.0 (active)
language-html, v0.46.1 (active)
language-hyperlink, v0.16.1 (active)
language-java, v0.24.0 (active)
language-javascript, v0.122.0 (active)
language-json, v0.18.3 (active)
language-less, v0.29.6 (active)
language-make, v0.22.2 (active)
language-mustache, v0.13.0 (active)
language-objective-c, v0.15.1 (active)
language-perl, v0.37.0 (active)
language-php, v0.37.3 (active)
language-property-list, v0.8.0 (active)
language-python, v0.45.1 (active)
language-ruby, v0.70.2 (active)
language-ruby-on-rails, v0.25.1 (active)
language-sass, v0.57.0 (active)
language-shellscript, v0.23.0 (active)
language-source, v0.9.0 (active)
language-sql, v0.25.0 (active)
language-text, v0.7.1 (active)
language-todo, v0.29.1 (active)
language-toml, v0.18.1 (active)
language-xml, v0.34.12 (active)
language-yaml, v0.27.1 (active)

# Dev
No dev packages
```

**Comment by xiongxiong** [comment]
use 'alt + cmd + f' to replace text in buffer

**Comment by xiongxiong** [comment]
and why the text to be replaced in the 'replace' text field?

**Comment by 50Wliu** [comment]
Is this reproducible in safe mode (`atom --safe`)?

**Comment by xiongxiong** [comment]
yes, reproducible in safe mode.

**Comment by Ben3eeE** [comment]
What keyboard layout are you using?

/cc: @nathansobo

**Comment by xiongxiong** [comment]
Sorry, I find it not a bug of Atom, it's a bug with input method(搜狗输入板), when I close it, everything is  ok.

**Comment by xiongxiong** [comment]
macbook pro, US keyboard, QWERTY

**Comment by nathansobo** [comment]
@xiongxiong How can I access that input method? Is it something built into the operating system or do I need to install it from elsewhere. It seems like the `KeyboardEvent.key` field is showing up as undefined somehow.

**Comment by xiongxiong** [comment]
I am a chinese user of Atom, you can search the input method "搜狗输入板" in App Store (China), but i don't know whether it is in App Store (USA).

**Comment by zzjin** [comment]
Same error here. I can confirm that it is not caused by `搜狗输入法`, when you chose mac's build in chinese input method, same error appeared.

Atom Version: 1.13.0-beta0
System: Mac OS X 10.12.1

**Comment by pdxwolfy** [comment]
This also occurs with the "Grammarian" input method from  Grammarian Pro. Switching to US or US International - PC and the problem goes away, at least until I reactivate the Grammarian keymap. This is new behavior with Atom Version 1.12.0 which the system installed last night. macOS version is 10.12.1. Grammarian Pro version is 2.0.11

**Comment by 50Wliu** [comment]
Cannot reproduce using Microsoft Pinyin, assuming macOS-only.

**Comment by kyoh86** [comment]
I'm Japanese user and met same error with the Google IME.
This is occurred in using `alt-cmd(-shift)?-(any)` formed keybindings like the `alt-cmd-o (advanced-open-file:toggle)`.

**Comment by nathansobo** [comment]
Sorry about this. We called the wrong function in the Carbon API and are not successfully retrieving the native keyboard layout when an input method such as Pinyin Simplified is in use. This slipped through our beta channel but we'll have a hotfix out soon.

**Comment by pdxwolfy** [comment]
Great!  Thanks, nathansobo.

**Comment by nathansobo** [comment]
Fixed via https://github.com/atom/atom-keymap/pull/170 and available in 1.12.1. Thanks for your reports and sorry about this regression.
